src/managers/queue-manager.js

```diff
@@ -108,17 +70,21 @@ function handleMessage(message) {
 
 export async function pollMessageQueue() {
   if (getUserToken()) {
-    var response = await getUserQueue();
-    if (response && (response.status === 200 || response.status === 204)) {
-      log(`Message queue checked for user ${getUserToken()}, ${response.data.length} messages found.`);
-      if (response.data.length > 0) {
-        messages = response.data;
-        checkMessageQueue();
+    if (Gist.isDocumentVisible) {
```

Comment:
Not necessary for this change, but this has me thinking a nice improvement could be to poll immediately whenever the document becomes visible? https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event